tests/page-objects/rerences-types.ts

```diff
@@ -0,0 +1,149 @@
+export type Names = {
+  lastName: string;
+  firstName: string;
+}[];
+
+export type CommonReferenceProperties = {
+  type: string;
```

Comment:
You could consider using an enum here, as you are string checking these in a couple of locations hence it would help maintainability.

```
enum ReferenceType {
  JOURNAL_ARTICLE = 'Journal Article',
  BOOK = 'Book'
  ...
}

export type CommonReferenceProperties = {
  type: ReferenceType;
  names: Names;
  doi?: string;
}

export type JournalReference = CommonReferenceProperties & {
  type: ReferenceType.JOURNAL_ARTICLE;
  ...
}

const reference = {
      type: ReferenceType.JOURNAL_ARTICLE,
      ...
}
```

Note, I think I'm stuck in a C/C++ mindset where I always capitalise the values but you can also do `ReferenceType.JournalArticle`.

Reply:
I tried this and it turned into a rabbit hole of chaos so i will look into it again after working on the other tests.